substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -94,17 +94,246 @@ private PosixJavaNetSubstitutions() {
 @TargetClass(className = "java.net.PlainDatagramSocketImpl")
 @Platforms({Platform.LINUX.class, Platform.DARWIN.class})
 final class Target_java_net_PlainDatagramSocketImpl {
+    @Alias boolean connected;
 
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
     protected void bind0(int lport, InetAddress laddr) throws SocketException {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.bind0(int, InetAddress)");
+        //        jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        //        int fd;
+        //        int len = 0;
+        CIntPointer len_Pointer = StackValue.get(CIntPointer.class);
```

Comment:
You forgot the initialization of `len = 0`.  (That's a common mistake!)  You need
```
len_Pointer.write(0);
```

Reply:
Though, in this case, by reading much further, it looks like `len` is written before it is read.  But I don't want to have to do that analysis.  And I don't want to have to debug that if the code evolves to read `len` before it is written.